src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,22 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Check whether there are nullable nulls in at least one part of the foreign key.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            if ($schema->isNullable($field) && $entity->get($field) === null) {
+                return true;
+            }
```

Comment:
The idea is that for any ExistsIn constraint where any field that can be null is null means that the rule is allowed to pass on this condition alone.
As I have added in the example in the book if you want to check additional constraints you do it separatly.
E.g. 
```sql
site_id NOT NULL
language_id NOT NULL
parent_id NULL
```
The correct constraint setup is:
```php
$rules->add($rules->existsIn(
  ['site_id', 'language_id', 'parent_id'], 'ParentNodes', ['passingOnPartialNulls' => true]));
$rules->add($rules->existsIn(
  ['site_id', 'language_id'], 'ParentNodes'));
```
Does that make sense to you?